src/main/java/se/isselab/HAnS/metricsView/MetricsViewFactory.java

```diff
@@ -0,0 +1,149 @@
+    public void createToolWindowContent(@NotNull Project project, @NotNull ToolWindow toolWindow) {
+        createButton(project, toolWindow);
+        addContent(toolWindow);
+    }
+
+    private void createButton(Project project, ToolWindow toolWindow) {
+        triggerButton = new JButton("Refresh Metrics");
+        triggerButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                // Run the task in the background
+                ProgressManager.getInstance().run(new Task.Backgroundable(project, "Refreshing Metrics") {
+                    @Override
+                    public void run(@NotNull ProgressIndicator indicator) {
+                        refreshTableContent(project, toolWindow);
+                    }
+                });
+            }
+        });
+    }
+
+    private void addContent(ToolWindow toolWindow) {
+        contentPanel = new JPanel();
+        contentPanel.setLayout(new BorderLayout());
+
+        JPanel buttonPanel = new JPanel();
+        buttonPanel.add(triggerButton);
+        contentPanel.add(buttonPanel, BorderLayout.NORTH);
+
+        ContentFactory contentFactory = ContentFactory.getInstance();
+        Content content = contentFactory.createContent(contentPanel, "", false);
+        toolWindow.getContentManager().addContent(content);
+    }
+
+    private void refreshTableContent(Project project, ToolWindow toolWindow) {
+        DefaultTableModel tableModel = new DefaultTableModel(new Object[]{"Feature", "Scattering Degree", "Tangling Degree", "Line Count"}, 0) {
+            @Override
+            public Class<?> getColumnClass(int columnIndex) {
+                // Set appropriate classes for sorting
+                switch (columnIndex) {
+                    case 1: // Scattering Degree column
+                    case 2: // Tangling Degree column
+                    case 3: // Line Count column
+                        return Integer.class; // Use Integer class for numerical sorting
+                    default:
+                        return Object.class; // Default class
+                }
+            }
+        };
+        JBTable table = new JBTable(tableModel);
+
+        TableRowSorter<DefaultTableModel> sorter = new TableRowSorter<>(tableModel);
+        Comparator<Integer> integerComparator = Comparator.comparingInt(Integer::intValue);
+        sorter.setComparator(1, integerComparator); // Scattering Degree column
+        sorter.setComparator(2, integerComparator); // Tangling Degree column
+        sorter.setComparator(3, integerComparator); // Line Count column
+        table.setRowSorter(sorter);
+
+        FeatureService featureService = project.getService(FeatureService.class);
+        List<FeatureModelFeature> features = featureService.getFeatures();
+
+        List<FeatureModelFeature> root = featureService.getRootFeatures();
+
+        for (FeatureModelFeature feature : features) {
```

Comment:
I think this for should probably be broken out to it's own method to reduce the complexity of this method